018-cli-2.0/018-cli-2.0.md

```diff
@@ -0,0 +1,393 @@
+
+```sh
+appwrite functions createTag --functionId=[ID] --command=[COMMAND]
+```
+
+This corresponding appwrite issue will also need to be tackled as we implement this.
+https://github.com/appwrite/appwrite/issues/1316
+
+### 🟢 The `appwrite.json` File
+
+The next improvement is going to be along the lines of a `appwrite.json` file. Running `appwrite init` in the current directory will create a `appwrite.json` file and initialize the current directory with attributes necessary to communicate with a particular project.
+
+Consider this example
+```sh
+mkdir temp && cd temp
+
+# Initialize the current directory with an Appwrite project 
+appwrite init
+
+temp 
+└── appwrite.json
+```
+
+A `appwrite.json` file will be associated with one project only whose ID will be visible in the `project` attribute. 
+
+```json
+{
+    "project" : "",
+    "functions": [
+        {
+          "name" : "My awesome function 💪",
+          "path" : "./FunctionOne",
+          "command" : "python main.py",
+          "runtime": "python3.9",
+          // This section is useful when we allow cloud functions to be tested locally
+          "vars" :{
+            "KEY 1" : "VALUE 1",
+          }
+        },
+    ],
+    "hosting": []
+}
+```
+
+There are two scenarios here 
+* A developer would like to link the directory to an existing project
+
+```sh
+appwrite init
+
+Choose a project to link this directory to 
+⭕️ Project 1
+✅️ Project 2
+⭕️ Project 3
+```
+
+* A developer would like to create a new project while initializing this directory
+
+```sh
+appwrite init --new
+
+Give your project a name:
+Project X
+
+Creating Project X...
+
+Project X Created 👍
+```
+
+> We need to first add support to utopia-php/CLI to allow for single select and multi select commands to enable this feature.
+
+
+### 🟢 The `appwrite deploy` command
+
+The `appwrite deploy` command is a convenient wrapper around three main commands
+* Create Function
+* Create Tag
+* Update Tag
+
+```sh
+# Usage 1
+appwrite deploy functions
+Which functions would you like to deploy? 
+✅️ FunctionOne
+✅️ FunctionTwo
+⭕️ FunctionThree
+
+# Usage 2
+appwrite deploy functions --all
+# Deploys all the functions specified in appwrite.json
+
+# Future support for static hosting
+appwrite deploy site
+appwrite deploy site --all 
+```
+
+## Easier creation of cloud functions
+---
+
+We will introduce an `appwrite generate` command that does the following
+* Generate a function / static site template. 
+* Make an entry in the `appwrite.json` file. Throw an error if the file doesn't exist.
+
+Here are some example usages 
+```sh
+# Usage 1
+appwrite generate function --name="FunctionOne" --runtime=python-3.9
+# Creates a folder called FunctionOne with a template for a sample python function.
+temp
+├── appwrite.json
+└── FunctionOne
+    ├── .appwrite
+    ├── main.py
+    └── requirements.txt
+
+
+# Usage 2
+appwrite generate function --name="FunctionTwo" --runtime=node-16.0
```

Comment:
May be we can introduce one more parameter here `--template` which should be a link to an accessible github repository/branch so that devs can customize to use their own templates as well as our more advance templates, even our demos can be used as template easily using this feature?